Add fire_event mode for rfxtrx component

badele opened the pull request.

Commit:
mqtt light component

Commit:
pylint rework for light/mqtt

Commit:
fix a bug after the pylint rework

Commit:
rework for flake8 errors done

Commit:
two different demo lights on without RGB and one with RGB support.
and code cleanup more pylint aligned

Commit:
light/mqtt to .coveragerc

hexxter committed:
redesigned mqtt light an first steps with the unittest system

hexxter committed:
mqtt light test is working more test should be written

hexxter committed:
not working mqtt light unittest

leoc committed:
[Zwave] Add binary switch component

leoc committed:
[Zwave] Add type and genre to value filter

leoc committed:
[Zwave] Add light zwave component

leoc committed:
[Zwave] Add check for missing `discovery_service`

fabaff committed:
Add link to docs and update some docstrings

turnrye committed:
Initial implementation of mjpeg camera

turnrye committed:
Fixed style issue related to failing build

balloob committed:
Style fixes + rename honeywell

turnrye committed:
Changed camera.mjpeg to use Response and Closing; cleaned up a number of code-clarity issues near that

turnrye committed:
Fixed line length issues to make lint happy. Still bummed that I decreased test coverage :(

turnrye committed:
More lint fixes

turnrye committed:
Fixed indentations hopefully

balloob committed:
Clean up notifiy component

balloob committed:
Style fixes

balloob committed:
Update coveragerc

balloob committed:
Check global installed packages

fabaff committed:
Update link to docs (Jekyll 3 update)

fabaff committed:
Update link to docs (Jekyll 3 update)

balloob committed:
Update limitlessled with lists for colors

balloob committed:
Add dev info to frontend urls

balloob committed:
Another LimitlessLED color fix

balloob committed:
Upgrade frontend with about page

balloob committed:
Compile new version frontend

fabaff committed:
Add pushetta

fabaff committed:
Add pushetta

balloob committed:
Update frontend with new card style

fabaff committed:
Add pushetta notify platform

fabaff committed:
Use _LOGGER.error instead of _LOGGER.exception

leoc committed:
[Zwave] Use threading.Timer for value refresh delay

leoc committed:
[Zwave] Fix docstring

leoc committed:
[Zwave] Update HA state on value change

hexxter committed:
more unittests

hexxter committed:
now saved

hexxter committed:
check the default value only checkable local. I removed it.

hexxter committed:
more self.hass.pool.block_till_done()

hexxter committed:
next online unittest test ;)

tomduijf committed:
Fixes memory consumption issue

lorddaren committed:
Fixing bug when connecting to squeezebox and it is a float

fabaff committed:
Update file header and docstrings

persandstrom committed:
more robust and more logging

balloob committed:
Move core light test to correct dir

balloob committed:
Fix MQTT light test

fabaff committed:
Add link to docs

nkgilley committed:
add exception handling to generic camera requests function.

nkgilley committed:
Remove rgb color if it's not an rgb bulb.

nkgilley committed:
last PR was dumb. this fix is better.

badele committed:
Add pushbutton mode for rfxtrx

fabaff committed:
Improve error messages, use constants, and fix docstrings

fabaff committed:
Fix typo

nkgilley committed:
remove break

balloob commented:
Pushbutton support should not be interwoven with a light or switch component but should be a new component called pushbutton with an rfxtrx platform.

I will create a foundation for this platform once I'm able to ship the 0.8 release so you can add rfxtrx to it. Ok?

happyleavesaoc committed:
s20 switch support

happyleavesaoc committed:
fixed requirements

happyleavesaoc committed:
fixed names

happyleavesaoc committed:
rename platform to orvibo

balloob commented on homeassistant/components/light/rfxtrx.py:
```diff
@@ -22,12 +27,21 @@ def setup_platform(hass, config, add_devices_callback, discovery_info=None):
+                _LOGGER.info("Add %s rfxtrx.light", entity_info[ATTR_NAME])
+
+                # Check if in push button mode
+                pushbutton = entity_info[ATTR_PUSHBUTTON] \
```
Can we rename this to `fire_event`, it describes better what it will do.

Also, you can use the `get` method on a dict to specify a default value:
```python
fire_event = entity_info.get(ATTR_FIRE_EVENT, False)
```

balloob commented on homeassistant/components/light/rfxtrx.py:
```diff
@@ -71,6 +87,16 @@ def light_update(event):
 
+                # Fire event
+                if rfxtrx.RFX_DEVICES[entity_id].ispushbutton:
+                    rfxtrx.RFX_DEVICES[entity_id].hass.bus.fire(
```
You can use `hass` instead of `rfxtrx.RFX_DEVICES[entity_id].hass`

nkgilley committed:
remove unnecessary else

happyleavesaoc committed:
addressed comments

balloob committed:
Script: new attribute if can cancel

balloob committed:
Update to new version frontend

rmkraus committed:
Created updater component

balloob committed:
Have Notify platform install platform dependencies

balloob committed:
Log error if unable to install package

balloob committed:
Remove lib directory in version upgrade

rmkraus committed:
Added friendly name to updater component.

balloob committed:
Only delete lib dir in config upgrade if exists

balloob committed:
Remove unused import in notify

happyleavesaoc committed:
don't update state in turn_on/off

badele committed:
Fix rfxtrx state (no resend the RF signal)

leoc committed:
[Zwave] Add zwave polling interval configuration

leoc committed:
[Zwave] Add Meter command class

fabaff committed:
Update link to docs

tomduijf committed:
Initial commit of pushbullet enhancement

tomduijf committed:
pushbullet; styling and minor fixed before PR

tomduijf committed:
Pushbullet; styling, requirements, example

tomduijf committed:
Added comments

rmkraus committed:
Updating fronted to latest version.

rmkraus committed:
Implemented suggestions from Paulus for updater

1) Better Error handling when making PyPI requests.
2) More efficient event scheduling.
3) ENTITY_ID in constant
3) friendly_name from constant

balloob committed:
Add tests for version upgrade

rmkraus committed:
Updater suggestions from Paulus

1) Moved error checking into get_newest_version function.
2) Fixed import formatting mistake.

rmkraus committed:
Using ATTR_FRIENDLY_NAME in updater component.

rmkraus committed:
Fixed typo in updater entity attributes.

Left some quotes in there. My bad.

rmkraus committed:
Fixed hass daemon management

1) Changed signal to exit hass to SIGTERM
2) Updated initd script to send SIGTERM
3) Updated systemd script to never send SIGKILL.

tomduijf committed:
<type>.<name> split only on first separator

tomduijf committed:
Processed feedback from PR comments

tomduijf committed:
Pushbullet; code cleanup & better errors on config typos

rmkraus committed:
Added updater to set of default components

fabaff committed:
Update docstrings and link to docs

badele committed:
Rename pushbutton attribute to fire_event

balloob committed:
Version bump to 0.8.0

balloob commented:
Seems like you're having some [style violations](https://travis-ci.org/balloob/home-assistant/builds/91329168#L285-L294)

I'll take a look at the code later today

badele committed:
Update the code style (pylint & pyflake)

badele committed:
Add state & isfire_event property

balloob commented on homeassistant/components/light/rfxtrx.py:
```diff
@@ -94,17 +121,37 @@ def name(self):
+        """ Returns is the device must fire event"""
+        return self._isfire_event
+
+    @isfire_event.setter
```
There is no need for this setter, it is set in the constructor and never changed.

balloob commented on homeassistant/components/light/rfxtrx.py:
```diff
@@ -94,17 +121,37 @@ def name(self):
         return self._name
 
     @property
+    def isfire_event(self):
```
Can you rename this to `should_fire_event` ?

badele committed:
Rename fire_event into should_fire_event

badele committed:
Fix state property
